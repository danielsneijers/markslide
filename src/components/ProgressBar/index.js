// @flow
import React from 'react'
import CSS from './style.css'

export type Props = {
  offset: number
}

const ProgressBar = ({ offset }: Props) => {
  const style = { transform: `translateX(-${offset}%)` }

  return <span className={CSS.progressBar} style={style} />
}

export default ProgressBar
