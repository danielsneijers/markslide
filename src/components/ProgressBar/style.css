.progressBar {
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  height: 2px;
  width: 100%;
  background-color: blue;
  overflow: hidden;
  transition: transform .2s ease-out;
}
