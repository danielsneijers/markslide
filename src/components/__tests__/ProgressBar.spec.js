import React from 'react'
import { shallow } from 'enzyme'
import ProgressBar from 'components/ProgressBar'

describe('components/ProgressBar', () => {
  test('renders correctly', () => {
    const tree = shallow(<ProgressBar offset={-30} />)

    expect(tree).toMatchSnapshot()
  })
})
