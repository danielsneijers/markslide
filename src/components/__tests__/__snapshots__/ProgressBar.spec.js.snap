// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProgressBar renders correctly 1`] = `
<span
  className="progressBar"
  style={
    Object {
      "transform": "translateX(--30%)",
    }
  }
/>
`;
